server/src/test/java/org/opensearch/action/DynamicActionRegistryTests.java

```diff
@@ -89,22 +89,30 @@ public void testDynamicActionRegistryWithNamedRoutes() {
 
         assertTrue(registry.isActionRegistered("foo"));
         assertTrue(registry.isActionRegistered("bar"));
+
+        registry.unregisterDynamicRoute(r2);
+
+        assertTrue(registry.isActionRegistered("foo"));
+        assertFalse(registry.isActionRegistered("bar"));
     }
 
-    public void testDynamicActionRegistryRegisterAndUnregisterWithNamedRoutes() {
+    public void testDynamicActionRegistryWithNamedRoutesAndLegacyActionNames() {
         RestSendToExtensionAction action = mock(RestSendToExtensionAction.class);
         RestSendToExtensionAction action2 = mock(RestSendToExtensionAction.class);
-        NamedRoute r1 = new NamedRoute(RestRequest.Method.GET, "/foo", "foo");
-        NamedRoute r2 = new NamedRoute(RestRequest.Method.GET, "/bar", "bar");
+        NamedRoute r1 = new NamedRoute(RestRequest.Method.GET, "/foo", "foo", "cluster:admin/opensearch/abc/foo");
+        NamedRoute r2 = new NamedRoute(RestRequest.Method.GET, "/bar", "bar", "cluster:admin/opensearch/xyz/bar");
 
         DynamicActionRegistry registry = new DynamicActionRegistry();
         registry.registerDynamicRoute(r1, action);
         registry.registerDynamicRoute(r2, action2);
 
+        assertTrue(registry.isActionRegistered("cluster:admin/opensearch/abc/foo"));
+        assertTrue(registry.isActionRegistered("cluster:admin/opensearch/xyz/bar"));
+
         registry.unregisterDynamicRoute(r2);
 
-        assertTrue(registry.isActionRegistered("foo"));
```

Comment:
Why not keep these assertions in too?

Reply:
I just combined two tests into one. So that instead of 4 tests (2 each, with and without legacyActionName), we'll have 2 for better readability and since most of the code was reused.